Keen/EventCollection.cs

```diff
@@ -1,100 +1,158 @@
-                    .ConfigureAwait(continueOnCapturedContext: false);
-                var responseString = await responseMsg.Content.ReadAsStringAsync()
-                    .ConfigureAwait(continueOnCapturedContext: false);
-                dynamic response = JObject.Parse(responseString);
-
-                // error checking, throw an exception with information from the json 
-                // response if available, then check the HTTP response.
-                KeenUtil.CheckApiErrorCode(response);
-                if (!responseMsg.IsSuccessStatusCode)
-                    throw new KeenException("GetSchema failed with status: " + responseMsg.StatusCode);
-
-                return response;
+                throw new ArgumentNullException(nameof(keenHttpClientProvider),
+                                                "A KeenHttpClient provider must be provided.");
             }
+
+            if (string.IsNullOrWhiteSpace(prjSettings.KeenUrl) ||
+                !Uri.IsWellFormedUriString(prjSettings.KeenUrl, UriKind.Absolute))
+            {
+                throw new KeenException(
+                    "A properly formatted KeenUrl must be provided via Project Settings.");
+            }
+
+            var serverBaseUrl = new Uri(prjSettings.KeenUrl);
+            _keenHttpClient = keenHttpClientProvider.GetForUrl(serverBaseUrl);
+            _eventsRelativeUrl = KeenHttpClient.GetRelativeUrl(prjSettings.ProjectId,
+                                                               KeenConstants.EventsResource);
+
+            // TODO : It's possible we may want to change back to dynamically grabbing the keys
+            // from a stored IProjectSettings so client code can lazily assign keys. It creates a
+            // minor potential race condition, but will allow for scenarios like creating a
+            // KeenClient instance with only a master key in order to generate/acquire access keys
+            // to then set as the other keys. Otherwise a new KeenClient must be created or at
+            // least a new instance of the IEventCollection/IEvent/IQueries implementations.
+
+            _readKey = prjSettings.ReadKey;
+            _writeKey = prjSettings.WriteKey;
+            _masterKey = prjSettings.MasterKey;
         }
 
-        public async System.Threading.Tasks.Task DeleteCollection(string collection)
+
+        public async Task<JObject> GetSchema(string collection)
         {
-            using (var client = new HttpClient())
+            // TODO : So much of this code, both in the constructor and in the actual message
+            // dispatch, response parsing and error checking is copy/paste across Queries, Event
+            // and EventCollection everywhere we use KeenHttpClient. We could shove some of that
+            // into shared factory functionality (for the ctor stuff) and some of it into the
+            // KeenHttpClient (for the dispatch/response portions).
```

Comment:
Yeah I'm thinking we should do something more drastic around these design decisions before v1 while we can still get away with beta-ish version-to-version breaks in compatibility. I'd rather the interfaces be more about the functionality as leveraged by client code rather than the exact shape of the HTTP resources of the API.